state/txindex/kv/kv.go

```diff
@@ -197,25 +196,25 @@ func (txi *TxIndex) Search(q *query.Query) ([]*types.TxResult, error) {
 		}
 	}
 
-	results := make([]*types.TxResult, len(hashes))
-	i := 0
-	for _, h := range hashes {
-		results[i], err = txi.Get(h)
-		if err != nil {
-			return nil, errors.Wrapf(err, "failed to get Tx{%X}", h)
-		}
-		i++
-	}
+	//results := make([]*types.TxResult, len(hashes))
+	//i := 0
+	//for _, h := range hashes {
+	//	results[i], err = txi.Get(h)
+	//	if err != nil {
+	//		return nil, errors.Wrapf(err, "failed to get Tx{%X}", h)
+	//	}
+	//	i++
+	//}
 
 	// sort by height & index by default
-	sort.Slice(results, func(i, j int) bool {
-		if results[i].Height == results[j].Height {
-			return results[i].Index < results[j].Index
-		}
-		return results[i].Height < results[j].Height
-	})
-
-	return results, nil
+	//sort.Slice(results, func(i, j int) bool {
+	//	if results[i].Height == results[j].Height {
+	//		return results[i].Index < results[j].Index
+	//	}
+	//	return results[i].Height < results[j].Height
+	//})
+
+	return hashes, nil
```

Comment:
We stored the key order by `Height` and `Index` , so not sort again.
https://github.com/irisnet/tendermint/blob/master/state/txindex/kv/kv.go#L44

Reply:
It's will order by tag1,tag2,tag3... height, index.